### PR TITLE
ceph-volume/cephadm: support lv devices in inventory

### PR DESCRIPTION
**Pull request by guits**
The inventory should report LV devices as they are valid devices that
can be used to be prepared as OSDs.
    
Fixes: https://tracker.ceph.com/issues/62362
    
Signed-off-by: Guillaume Abrioux <gabrioux@ibm.com>

### CODE REVIEW

File: src/ceph-volume/ceph_volume/api/lvm.py [hunk 1]
```diff
@@ -1209,3 +1210,39 @@ def get_lv_by_fullname(full_name):
...
+    """
+    This functions translates a given lv path under the format:
+    /dev/VG/LV to the format /dev/mapper/LV.
+    eg:
+    from:
+    /dev/ceph-c1a97e46-234c-46aa-a549-3ca1d1f356a9/osd-block-32e8e896-172e-4a38-a06a-3702598510ec
+    to:
+    /dev/mapper/ceph--c1a97e46--234c--46aa--a549--3ca1d1f356a9-osd--block--32e8e896--172e--4a38--a06a--3702598510ec
+    """
+    results = re.split(r'^\/dev\/(.+\w)-(\w.+)', lv_path)
```
Comment (adk3798) [inline src/ceph-volume/ceph_volume/api/lvm.py:1242]: These functions going mapper -> LV and LV -> mapper names seem like they would be nice functions to have unit tests for given they look like pure functions.

**Review by adk3798 — APPROVED** [review]
LGTM, a couple nitpicks but I think this is okay even if you disagree and don't address them.

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -165,6 +165,12 @@ class CephadmOrchestrator(orchestrator.Orchestrator, MgrModule,
             default=False,
             desc='Use libstoragemgmt during device scans',
         ),
+        Option(
+            'inventory_list_all',
```
[the remark this answers is not in the document]
Comment (guits) [reply src/pybind/mgr/cephadm/module.py:169]: I wanted something a bit more generic because it might be not only about lv devices at the end.

**Comment by guits** [comment]
@adk3798 I think it's missing a `self.inventory_list_all = False` in the `__init__()` of `class CephadmOrchestrator()`, right?

File: src/pybind/mgr/cephadm/module.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (guits) [reply src/pybind/mgr/cephadm/module.py:171]: fwiw, I've updated the `desc`

**Comment by adk3798** [comment]
> ~@adk3798 I think it's missing a `self.inventory_list_all = False` in the `__init__()` of `class CephadmOrchestrator()`, right?~

ah, yeah. I had forgot about that. It's only necessary so mypy can work I think.

File: src/pybind/mgr/cephadm/module.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adk3798) [reply src/pybind/mgr/cephadm/module.py:171]: okay, makes sense if it's more than just LVs

**Comment by guits** [comment]
https://pulpito.ceph.com/gabrioux-2023-08-10_11:37:44-orch:cephadm-wip-guits-testing-5-2023-08-10-0921-distro-default-smithi/

**Comment by guits** [comment]
https://pulpito.ceph.com/gabrioux-2023-08-10_21:20:10-orch:cephadm-wip-guits-testing-5-2023-08-10-1324-distro-default-smithi/

**Review by asm0deuz — APPROVED** [review]
LGTM

**Comment by adk3798** [comment]
> https://pulpito.ceph.com/gabrioux-2023-08-10_21:20:10-orch:cephadm-wip-guits-testing-5-2023-08-10-1324-distro-default-smithi/

All failures in that run were known issues FYI. Looked pretty clean to me.

**Pull request merged**